### PR TITLE
log add & init func

### PR DESCRIPTION
**Pull request by vladislavprovich**

### CODE REVIEW

File: internal/service/url_service.go [hunk 1]
```diff
@@ -83,34 +83,53 @@ func (s *urlService) CreateShortURL(ctx context.Context, req models.ShortenReque
...
 func (s *urlService) GetOriginalURL(ctx context.Context, shortURL string) (string, error) {
+	s.logger.Info("service.GetOriginalURL", zap.String("short_url", shortURL))
 	originalUrl, err := s.repo.GetURL(ctx, shortURL)
 	if err != nil {
+		s.logger.Info("service, failed to get original URL", zap.String("short_url", shortURL))
 		return "", fmt.Errorf("get short url, get url err:, %s", err)
 	}
 	// Check if URL has expired
 	if originalUrl.ExpiredAt != nil && time.Now().After(*originalUrl.ExpiredAt) {
+		s.logger.Info("storage time has expired, URL has expired", zap.String("short_url", shortURL))
```
Comment (vkotsiuba99) [inline internal/service/url_service.go]: storage?

File: internal/handler/url_handler.go [hunk 2]
```diff
@@ -87,26 +91,34 @@ func (h *URLHandler) ShortenURL(w http.ResponseWriter, r *http.Request) {
...
 		http.Error(w, "URL not found", http.StatusNotFound)
 		return
 	}
 
 	// Log the redirect
 	referrer := r.Referer()
-	h.service.LogRedirect(r.Context(), shortURL, referrer)
-
+	h.logger.Info("handler, referrer ", zap.String("referrer", referrer))
+	err2 := h.service.LogRedirect(r.Context(), shortURL, referrer)
```
Comment (vkotsiuba99) [inline internal/handler/url_handler.go]: if shadowed declaration
err = h....

File: internal/handler/url_handler.go [hunk 3]
```diff
@@ -47,8 +47,12 @@ func InitDB(connStr string) (*sql.DB, error) {
 
 func (h *URLHandler) ShortenURL(w http.ResponseWriter, r *http.Request) {
 	h.logger.Info("handler.ShortenURL called")
+	h.logger.Debug("handler.ShortenURL called")
 	var req models.ShortenRequest
 	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+
+		h.logger.Debug("handler, failed to decode request body", zap.Error(err))
```
Comment (vkotsiuba99) [inline internal/handler/url_handler.go]: not needed, if you debug something, debug it on a local branch or in another PR

Comment (vkotsiuba99) [reply internal/handler/url_handler.go]: already got same line 
h.logger.Debug("handler, failed to decode request body", zap.Error(err))
h.logger.Error("handler, failed to decode request body", zap.Error(err))

File: cmd/server/main.go [hunk 4]
```diff
@@ -61,24 +60,57 @@ func main() {
...
+	db, err := handler.InitDB(os.Getenv("DATABASE_URL"))
+	if err != nil {
+		log.Fatal("Failed to connect to database", zap.Error(err))
+	}
+	defer db.Close()
+}
+
+func initRepo() {
+	repo := repository.NewURLRepository(db)
+	fmt.Println(repo)
```
Comment (vkotsiuba99) [inline cmd/server/main.go]: remove all Println in init, you must return values from init func

File: cmd/server/main.go [hunk 5]
```diff
@@ -61,24 +60,57 @@ func main() {
...
+}
+
+func initService() {
```
Comment (vkotsiuba99) [inline cmd/server/main.go]: func initService() {
return service.NewURLService(*repo, log)
}

Comment (vkotsiuba99) [reply cmd/server/main.go]: fix all init

File: cmd/server/main.go [hunk 6]
```diff
@@ -15,32 +19,27 @@ import (
 	"go.uber.org/zap"
 )
 
+var log *zap.Logger
+var db *sql.DB
+var repo *repository.URLRepository
```
Comment (vkotsiuba99) [inline cmd/server/main.go]: don't init vars like this

Comment (vkotsiuba99) [reply cmd/server/main.go]: must be in main

repo := initRepo()

File: cmd/server/main.go [hunk 7]
```diff
@@ -15,32 +19,27 @@ import (
...
+	r.Use(middleware.Recoverer(log))
+	r.Use(middleware.RequestLogger(log))
 	r.Use(middleware.CORS)
 	r.Use(middleware.RateLimiter)
 
 	// TODO repository and service must init here together with urlHandler
 	// Initialize Handlers
-	urlHandler := handler.NewURLHandler(db, logger)
-
+	urlHandler := handler.NewURLHandler(db, log)
```
Comment (vkotsiuba99) [inline cmd/server/main.go]: all init must be here

File: cmd/server/main.go [hunk 8]
```diff
@@ -15,32 +19,27 @@ import (
...
-	}
-	defer db.Close()
-
 	// Create Router
 	r := chi.NewRouter()
 
 	// Apply Middlewares
-	r.Use(middleware.Recoverer(logger))
-	r.Use(middleware.RequestLogger(logger))
+	r.Use(middleware.Recoverer(log))
```
Comment (vkotsiuba99) [inline cmd/server/main.go]: better to keep "logger" than "log"

File: cmd/server/main.go (hunk 7 above)
Comment (vkotsiuba99) [inline cmd/server/main.go]: fix handler.NewURLHandler it's create new repo and storage but we created it in main

File: internal/service/url_service.go (hunk 1 above)
Comment (vladislavprovich) [reply internal/service/url_service.go]: storing urls in the database, it seems to be 30 days for us

File: internal/repository/init_db.go [hunk 9]
```diff
@@ -0,0 +1,16 @@
+package repository
+
```
Comment (vkotsiuba99) [inline internal/repository/init_db.go]: rename file to postgres
init_db -> postgres

File: cmd/server/main.go [hunk 10]
```diff
@@ -82,3 +92,31 @@ func main() {
 
 	logger.Info("Server exited gracefully")
 }
+
+func initLoger() *zap.Logger {
+	// Initialize Logger
```
Comment (vkotsiuba99) [inline cmd/server/main.go]: remove this comment, everything understand from func name

File: cmd/server/main.go [hunk 11]
```diff
@@ -82,3 +92,31 @@ func main() {
...
+	return logger.NewLogger(os.Getenv("LOG_LEVEL"))
+
+}
+
+func initDB() *sql.DB {
+	//Connect to Database
```
Comment (vkotsiuba99) [inline cmd/server/main.go]: remove this comment, everything understand from func name

**Pull request merged**